amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/DataFetcher.java

```diff
@@ -0,0 +1,44 @@
+package software.amazon.kinesis.retrieval.polling;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import lombok.NonNull;
+import software.amazon.awssdk.services.kinesis.model.GetRecordsRequest;
+import software.amazon.awssdk.services.kinesis.model.GetRecordsResponse;
+import software.amazon.awssdk.services.kinesis.model.GetShardIteratorRequest;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.common.StreamIdentifier;
+import software.amazon.kinesis.retrieval.DataFetcherResult;
+import software.amazon.kinesis.retrieval.kpl.ExtendedSequenceNumber;
+
+public interface DataFetcher {
+
+    DataFetcherResult getRecords();
+
+    void initialize(String initialCheckpoint,
+                    InitialPositionInStreamExtended initialPositionInStream);
+
+    void initialize(ExtendedSequenceNumber initialCheckpoint,
+                    InitialPositionInStreamExtended initialPositionInStream);
+
+    void advanceIteratorTo(String sequenceNumber,
+                           InitialPositionInStreamExtended initialPositionInStream);
+
+    void restartIterator();
+
+    void resetIterator(String shardIterator,
+                       String sequenceNumber,
+                       InitialPositionInStreamExtended initialPositionInStream);
+
+    boolean isShardEndReached();
+
+    GetRecordsResponse getResponse(GetRecordsRequest request) throws ExecutionException, InterruptedException, TimeoutException;
+
+    GetRecordsRequest getRequest(@NonNull String nextIterator) throws ExecutionException, InterruptedException, TimeoutException;
```

Comment:
Does these types GetRecordsRequest and GetRecordsResponse fit for archiver read only client as well? If not then we need generics here,

```
public interface DataFetcher<Req, Resp> {
...
Resp getResponse(Req request);
Req getRequest(String nextIterator);
Resp getRecords(String nextIterator);
...
}
```

check if we need to do this for ```String getNextIterator(GetShardIteratorRequest request)``` as well

Reply:
I created methods to convert the requests and responses to types that Archiver can use